### PR TITLE
Enhance front end documentation

### LINKED ISSUE #31: Screenshot as teaser
It would be cool to have a screenshot on top of the readme so that a user can immediately guess what the plugin is made for and does not have to read text for this.

### PR DESCRIPTION
**Pull request by jo-mueller**
Just some additional screenshots and a gif on the front page about what can currently be done with the plugin.

Fixes #31

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # napari-stress
 
-This plugin allows to segment fluorescence-labelled droplets , determine the surface with a ray-casting approach and calculate the surface curvatures. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made public in [this repository](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1).
+This plugin provides tools for the analysis of surfaces in Napari. It provides utilies to determine and refine the surface-representations of objects with a ray-casting approach and calculate surface curvatures. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made public in [this repository](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1).
+
```
Comment (Cryaaa) [inline README.md:4]: The second link to: "[this repository]" is the same link to the paper, I'm not sure that's right

File: README.md [hunk 2]
```diff
@@ -27,24 +29,29 @@ pip install -e .
...
+### Plugins
+
+All functions in this repository are documented sesparately as Jupyter notebooks [here](./docs/notebooks/demo). 
 
-* Confocal data (*.tif*), 3D+t: This type of data can be processed with napari-stressed as show in [this notebook]([https://github.com/BiAPoL/napari-stress/blob/split-up-main-widget/docs/notebooks/Process_confocal.ipynb](https://github.com/BiAPoL/napari-stress/blob/add-timelapse-decorator-for-points-and-surfaces/docs/notebooks/Process_confocal.ipynb))
-* Lightsheet data (*.czi*), 3D + t: coming soon....
+### Utilities
 
-The resulting surface will look like this:
+Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers some convenient way to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/add-timelapse-decorator-for-points-and-surfaces/docs/notebooks/demo/TimeLapse_processing.ipynb).
```
Comment (Cryaaa) [inline README.md:51]: The link added here is broken. I get a 404 not found error

**Review by Cryaaa — COMMENTED** [review]
So I think the documentation is definitely good for an introduction. I think that some of the notebooks that are mentioned in the readme could possibly contain more text, but I don't have nay background knowledge on any of the functions. One think that I noticed is that the spherical harmonics notebook has some cells not shown that were run (maybe a re-run of it is needed).

File: README.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # napari-stress
 
-This plugin allows to segment fluorescence-labelled droplets , determine the surface with a ray-casting approach and calculate the surface curvatures. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made public in [this repository](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1).
+This plugin provides tools for the analysis of surfaces in Napari. It provides utilies to determine and refine the surface-representations of objects with a ray-casting approach and calculate surface curvatures. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made public in [this repository](https://github.com/campaslab/STRESS).
```
Comment (haesleinhuepf) [inline README.md:3]:
```suggestion
This plugin provides tools for the analysis of surfaces in Napari, such as utilities to determine and refine the surface-representation of objects using a ray-casting approach and calculate the curvature of surfaces. It re-implements code in Napari that was written for [Gross et al. (2021): STRESS, an automated geometrical characterization of deformable particles for in vivo measurements of cell and tissue mechanical stresses](https://www.biorxiv.org/content/10.1101/2021.03.26.437148v1) and has been made open source in [this repository](https://github.com/campaslab/STRESS).

```

File: README.md [hunk 4]
```diff
@@ -27,24 +29,29 @@ pip install -e .
...
 
-### General timelapse-processing
-
-Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers some convenient way to extent other function's functionality (which are often made for 3D data) to timelapse data using the `frame_by_frame` function and the `TimelapseConverter` class, both of which are described in more detail in [this notebook]([url](https://github.com/BiAPoL/napari-stress/blob/add-timelapse-decorator-for-points-and-surfaces/docs/notebooks/demo/TimeLapse_processing.ipynb)).
+Functionality in this repository is divided in two groups: **Recipes** and **plugins**.
 
 ### Recipes
 
-Napari-stress provides jupyter notebooks with complete workflows for different types of input data. Napari-stress currently provides notebooks for the following data/image types:
+Napari-stress provides jupyter notebooks with [complete workflows](./docs/notebooks/recipes) for different types of input data and objectives. Napari-stress currently provides notebooks for the following data/image types:
```
Comment (haesleinhuepf) [inline README.md:36]: I suggest to have napari user-documentation first and code second.

File: README.md [hunk 5]
```diff
@@ -27,24 +29,29 @@ pip install -e .
...
+
+* **Confocal data** (*.tif*), 3D+t: This type of data can be processed with napari-stressed as show in [this notebook](./docs/notebooks/recipes/Process_confocal.ipynb)
+
+![](./docs/imgs/confocal/workflow.png)
+
+
+* **Lightsheet data** (*.czi*), 3D + t: coming soon....
```
Comment (haesleinhuepf) [inline README.md:43]: If it's coming soon, remove it from the readme and create a github issue instead.

File: README.md [hunk 6]
```diff
@@ -27,24 +29,29 @@ pip install -e .
...
+
+### Plugins
+
+All functions in this repository are documented sesparately as Jupyter notebooks [here](./docs/notebooks/demo).
```
Comment (haesleinhuepf) [inline README.md:47]:
```suggestion
All functions in this repository are documented separately as Jupyter notebooks [here](./docs/notebooks/demo). 

```

File: README.md [hunk 7]
```diff
@@ -27,24 +29,29 @@ pip install -e .
...
+### Plugins
+
+All functions in this repository are documented sesparately as Jupyter notebooks [here](./docs/notebooks/demo). 
 
-* Confocal data (*.tif*), 3D+t: This type of data can be processed with napari-stressed as show in [this notebook]([https://github.com/BiAPoL/napari-stress/blob/split-up-main-widget/docs/notebooks/Process_confocal.ipynb](https://github.com/BiAPoL/napari-stress/blob/add-timelapse-decorator-for-points-and-surfaces/docs/notebooks/Process_confocal.ipynb))
-* Lightsheet data (*.czi*), 3D + t: coming soon....
+### Utilities
 
-The resulting surface will look like this:
+Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers some convenient way to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/main/docs/notebooks/demo/demo_timelapse_processing.ipynb).
```
Comment (haesleinhuepf) [inline README.md:51]:
```suggestion
Data to be used for this plugin is typically of the form `[TZYX]` (e.g., 3D + time). Napari-stress offers convenient ways to use functions from other repositories (which are often made for 3D data) on timelapse data with the `frame_by_frame` function and the `TimelapseConverter` class. Both are described in more detail in [this notebook](https://github.com/BiAPoL/napari-stress/blob/main/docs/notebooks/demo/demo_timelapse_processing.ipynb).

```

**Review by haesleinhuepf — CHANGES_REQUESTED** [review]
Hi Johannes,

just some suggestions from my side. Furthermore, one request: Could you please write a short tutorial (without jupyter notebooks) into the readme telling users how to use basic functionality of napari-stress from menus in napari and/or by clicking instead of coding? Non-coding users will appreciated this.

Thanks!
Robert

**Comment by jo-mueller** [comment]
Hi @haesleinhuepf 

> just some suggestions from my side. Furthermore, one request: Could you please write a short tutorial (without jupyter notebooks) into the readme telling users how to use basic functionality of napari-stress from menus in napari and/or by clicking instead of coding? Non-coding users will appreciated this.

This is a good idea, the grant proposal will require this sort of documentation anyway. I think I'll add another folder to the `docs` with something like point and click documentation for each function.

**Comment by jo-mueller** [comment]
I added a bit of code to provide sample data, I hope that this makes it a bit easier for users to follow the instructions.

**Comment by haesleinhuepf** [comment]
Hi Johannes @jo-mueller ,

awesome, it already looks much better now. Still, I must say that as a new user, I would not know how to use it.  The [only tutorial](https://github.com/BiAPoL/napari-stress/blob/91ea05ed3c1938b75e97b58177b703e5bc7cdb93/docs/tutorials/point_and_click/demo_spherical_harmonics.md) starts with "open your own data" and it is not specified what data format that must be. Is it a raw image, a binary image, a label image, a point cloud?

It would be fantastic if you could describe a workflow how, I as an end-user who cannot code, can go from an image of a blob to a point cloud and then make use of napari-stress to measure curvature:
1. How to open an image
2. How to make a point cloud
3. How to turn the point cloud into a surface
4. How to measure the curvature of the surface

This very simple workflow is the bare minimum necessary to convince users that this is a cool tool. napari-stress does provide functionality to create point clouds from raw images, correct?

Thanks!

Best,
Robert

**Comment by jo-mueller** [comment]
Hi Robert @haesleinhuepf 

Thanks for the feedback!

> napari-stress does provide functionality to create point clouds from raw images, correct?

Yes. But the workflow requires a bunch of functions (surface reconstruction, surface tracing, smoothing) which are on different PRs (PR#38, #39, #40).

I think I'd rather keep this PR a little smaller to fix the linked issue on the top and instead create a new issue for the missing recipe.

**Comment by haesleinhuepf** [comment]
Feel free to merge @jo-mueller 

Looking forward to seeing everything together with nice documentation

**Pull request merged**